lib/src/validators/collection_validators.dart

```diff
@@ -0,0 +1,311 @@
+/// [minLengthMsg], returns the default localized error message from
+/// `FormBuilderLocalizations.current.minLengthErrorText(min)`.
+///
+/// ## Type Parameters
+/// - `T`: The type of input to validate. Must extend `Object` and can be a
+///   `String`, `Iterable`, `Map`, or other object type.
+///
+/// ## Parameters
+/// - `min` (`int`): The minimum length required. Must be non-negative.
+/// - `minLengthMsg` (`String Function(T input, int min)?`): Optional
+///   function to generate custom error messages. Receives the input and the
+///   minimum length required and returns an error message string.
+///
+/// ## Returns
+/// Returns a `Validator<T>` function that takes an input of type T and returns:
+/// - `null` if the input's length is greater than or equal to [min]
+/// - An error message string if the validation fails
+///
+/// ## Throws
+/// - `AssertionError`: When [min] is negative
+///
+/// ## Examples
+/// ```dart
+/// // String validation
+/// final stringValidator = minLength<String>(3);
+/// print(stringValidator('abc')); // Returns null
+/// print(stringValidator('ab')); // Returns error message
+///
+/// // List validation
+/// final listValidator = minLength<List>(2);
+/// print(listValidator([1, 2, 3])); // Returns null
+/// print(listValidator([1])); // Returns error message
+///
+/// // Custom error message
+/// final customValidator = minLength<String>(
+///   5,
+///   minLengthMsg: (_, min) => 'Text must be at least $min chars long!',
+/// );
+/// ```
+///
+/// ## Caveats
+/// - Non-collection objects (those that are neither String, Iterable, nor Map)
+///   are treated as having a length of 1
+/// {@endtemplate}
+Validator<T> minLength<T extends Object>(int min,
+    {String Function(T input, int min)? minLengthMsg}) {
+  assert(min >= 0, 'The "minLength" parameter may not be negative');
+  return (T input) {
+    // I think it makes more sense to say that scalar objects has length 1 instead of 0.
+    int valueLength = 1;
+
+    if (input is String) {
+      valueLength = input.length;
+    } else if (input is Iterable) {
+      valueLength = input.length;
+    } else if (input is Map) {
+      valueLength = input.length;
+    }
+
+    return valueLength < min
+        ? minLengthMsg?.call(input, min) ??
+            FormBuilderLocalizations.current.minLengthErrorText(min)
+        : null;
+  };
+}
+
+/// {@template validator_max_length}
+/// Creates a validator function that checks if the input's length is less than
+/// or equal to `max`. The validator returns `null` for valid input and an
+/// error message for invalid input.
+///
+/// If validation fails and no custom error message generator is provided via
+/// [maxLengthMsg], returns the default localized error message from
+/// `FormBuilderLocalizations.current.maxLengthErrorText(max)`.
+///
+/// ## Type Parameters
+/// - `T`: The type of input to validate. Must extend `Object` and can be a
+///   `String`, `Iterable`, `Map`, or other object type.
```

Comment:
Could be another object type? Make sense this?
If validator recieve for example an int, never will validate a `valueLength` of object.